protocols/kad/src/behaviour.rs

```diff
@@ -352,6 +356,19 @@ impl KademliaConfig {
         self.caching = c;
         self
     }
+
+    /// Enable or disable the `client` mode of a node.
+    ///
+    /// Client mode is disabled by default.
+    ///
+    /// In `client` mode, the peer would not be added to any other peer's 
+    /// routing table. The client peer can send queries over the Kademlia
+    /// network, but will not receive any queries from the other peer's in
+    /// the network.
+    pub fn set_client_mode(&mut self, t: bool) -> &mut Self {
```

Comment:
```suggestion
    pub fn set_mode(&mut self, t: bool) -> &mut Self {
```

I would talk about the `Mode` in general, instead of the client mode in specific. The method would take a `Mode`, see comment above.